### DISCUSSION TITLE
Errors in GPS matching when large number of observations or small amount of trimming

### DISCUSSION THREAD

**Discussion by m-qin** [NSAPH-Software/CausalGPS#143] [Q&A] [answered]
I'm running generate_pseudo_pop() on a random subset of my full dataset and get the following two errors when (a) I use a large subset of the data or (b) when I use less trimming in the "trim_quantiles" parameter. Would appreciate any advice!

Error in xgboost::xgb.DMatrix(data = X, label = Y, weight = obsWeights) : 
  [16:20:52] amalgamation/../src/data/data.cc:1163: Check failed: valid: Input data contains `inf` or `nan`

In addition: Warning message:
In FUN(X[[i]], ...) : Error in algorithm m_xgboost_internal 
  The Algorithm will be removed from the Super Learner (i.e. given weight 0) 

My code is the following:

```
n_random_rows <- 75000
random_rows <- sample(1:nrow(data), n_random_rows)
Y_subset <- data[random_rows, outcome_var]
w_subset <- data[random_rows, exposure_var]
c_subset <- as.data.frame(subset(data[random_rows,], select = confounders_with_categorical_as_factors))

matched_pop_subset <- generate_pseudo_pop(Y_subset,
                                 w_subset,
                                 c_subset,
                                  ci_appr = "matching",
                                  pred_model = "sl",
                                  gps_model = "parametric",
                                  use_cov_transform = TRUE,
                                  transformers = list("pow2", "pow3", "sqrt", "log"),
                                  sl_lib = c("m_xgboost"),
                                  params = list(xgb_nrounds = c(10, 20, 30, 50)),
                                  nthread = 15, # number of cores
                                  covar_bl_method = "absolute",
                                  covar_bl_trs = 0.1,
                                  covar_bl_trs_type = "maximal",
                                 optimized_compile = TRUE,
                                  trim_quantiles = c(0.05, 0.95),
                                  max_attempt = 5,
                                  matching_fun = "matching_l1",
                                  delta_n = 0.1,
                                  scale = 1)
```

**Comment by Naeemkh** [comment, marked as answer]
Hi @m-qin, your provided example is not reproducible. I do not have the data. However, it seems you are trying to take a log from a negative number.

**Reply by m-qin** [reply]
Thanks so much—yes, the issue was that I was trying to take log of a variable that sometimes equals 0! I've written a new function that takes the log of {a variable + 0.001} and am using that instead. There's no error anymore!
